unittest/mysys/my_vsnprintf-t.c

```diff
@@ -107,6 +109,49 @@ int main(void)
   test1("Precision works for strings <ab...>",
         "Precision works for strings <%.5T>", "abcdef!");
 
+  // MDEV-21978, tests based on those for their previous incarnations
+
+  test1("MariaDB extension escape sS works",
+        "MariaDB extension escape sS %sS", "works");
+  test1("MariaDB extension escape uU 2",
+        "MariaDB extension escape uU %uU", 2);
+
+  test1("MariaDB extension sQ works: `abcd` `op``q`",
+        "MariaDB extension sQ works: %sQ %.4sQ", "abcd", "op`qrst");
+
+  {
+    // Copied from ::test1
+    const char *res= "12034";
+    size_t len= my_snprintf(buf, sizeof(buf)-1, "\1%.3sB\4", "\2\0\3");
+    // Shift the chars so they (namely `\0`) are printable
+    for (unsigned int i= 0; i < len; ++i)
+      buf[i]+= '0';
+    ok(strlen(res) == len && strcmp(buf, res) == 0,
+       "\"MariaDB extension sB works: %s\"", buf);
+  }
```

Comment:
I copy-pasted a unique section to confirm that `%sB` ignores `\0`.
I also learned that `strncmp` stops at `\0` like `strcmp` does.

(Yeah, for the entire test file, I’ve only got half an idea on what I’m writing, just like the original author to some extent. 😅)

Reply:
`strncmp` should work with `NULL` terminated strings. But you will likely get a positive or negative result.